firesql/sql/sql_join.py

```diff
@@ -60,25 +60,34 @@ def inner_join(self, leftJoinPart: JoinPart, rightJoinPart: JoinPart) -> List:
     # create fast lookup for the right part
     keyLookup = {}
     for docId, doc in lookupPart.docs.items():
-      key = doc[lookupPart.joinField]
+      key = doc[lookupPart.joinField] if lookupPart.joinField != "docid" else docId
```

Comment:
This is a key piece in support for joining based on docid as the PK, and a FK in the joined table.